pkg/relay.go

```diff
@@ -150,50 +154,52 @@ func (rs *DefaultRelay) processValidator(ctx context.Context, payload []types.Si
 			return err
 		} else if !ok {
 			if rs.config.CheckKnownValidator {
-				return fmt.Errorf("not a validator")
+				return fmt.Errorf("%s not a known validator", registerRequest.Message.Pubkey.String())
 			} else {
-				logger.WithField("pubkey", pk.PublicKey).Debug("is not a known validator")
+				logger.
+					WithField("pubkey", pk.PublicKey).
+					WithField("slot", state.Beacon().HeadSlot()).
+					Debug("not a known validator")
 			}
 		}
 
 		// check previous validator registration
 		previousValidator, err := state.Datastore().GetRegistration(ctx, pk)
 		if err != nil && !errors.Is(err, ds.ErrNotFound) {
-			log.Warn(err)
+			logger.Warn(err)
 		}
 
 		if err == nil {
 			// skip registration if
 			if registerRequest.Message.Timestamp < previousValidator.Message.Timestamp {
-				rs.Log().Debug("request timestamp less than previous")
+				logger.WithField("pubkey", registerRequest.Message.Pubkey).Debug("request timestamp less than previous")
 				continue
 			}
 
 			if registerRequest.Message.Timestamp == previousValidator.Message.Timestamp &&
 				(registerRequest.Message.FeeRecipient != previousValidator.Message.FeeRecipient ||
 					registerRequest.Message.GasLimit != previousValidator.Message.GasLimit) {
 				// to help debug issues with validator set ups
-				rs.Log().With(log.F{
+				logger.With(log.F{
 					"prevFeeRecipient":    previousValidator.Message.FeeRecipient,
 					"requestFeeRecipient": registerRequest.Message.FeeRecipient,
 					"prevGasLimit":        previousValidator.Message.GasLimit,
 					"requestGasLimit":     registerRequest.Message.GasLimit,
+					"pubkey":              registerRequest.Message.Pubkey,
 				}).Debug("different registration fields")
 			}
 		}
 
 		// officially register validator
 		if err := state.Datastore().PutRegistration(ctx, pk, registerRequest, rs.config.TTL); err != nil {
-			rs.Log().WithError(err).Debug("Error in PutRegistration")
-			return err
+			logger.WithField("pubkey", registerRequest.Message.Pubkey).WithError(err).Debug("Error in PutRegistration")
+			return fmt.Errorf("failed to store %s", registerRequest.Message.Pubkey.String())
 		}
-
-		logger.WithField("pubkey", pk).Trace("validator registered")
 	}
 
 	logger.With(log.F{
-		"processingTimeMs": time.Since(timeStart).Milliseconds(),
-		"numberValidators": len(payload),
+		"processingTimeMs":      time.Since(timeStart).Milliseconds(),
+		"numberBatchValidators": len(payload),
```

Comment:
I don't think this is necessary for the same reason as above, mainly we can discern this is for a batch by the message